samtranslator/model/apigateway.py

```diff
@@ -99,7 +101,12 @@ def make_auto_deployable(self, stage, openapi_version=None, swagger=None, domain
             hash_input.append(str(openapi_version))
         if domain:
             hash_input.append(json.dumps(domain))
-
+        if redeploy_restapi_parameters:
+            function_names = redeploy_restapi_parameters.get('function_names')
+        else:
+            function_names = None
+        if function_names and function_names.get(self.logical_id[:-10], None):
```

Comment:
Can you remind me again why `self.logical_id[:-10]` is needed here? Maybe add a comment about what you're trying to do?

Reply:
Sure. I will add. self.logical_id has the api_name and the word "deployment". I am removing the word "deployment" and just getting the api_name to search for corresponding string of function_name/names